### PR TITLE
issue #1904: GmlReader ignores srsDimension attribute in the posList

### PR DESCRIPTION
**Pull request by xuzhg**
### Issues

*This pull request fixes issue #1904:.*

### Description

* Enable GmlReader to read the srsDimension.

### Checklist (Uncheck if it is not completed)

- [ ] *Test cases added*
- [ ] *Build and test with one-click build and test script passed*

### Additional work necessary

*If documentation update is needed, please add "Docs Needed" label to the issue and provide details about the required document change in the issue.*

### CODE REVIEW

File: test/FunctionalTests/Microsoft.Spatial.Tests/GmlReaderTests.cs [hunk 1]
```diff
@@ -26,6 +26,8 @@ public void ReadCoordinateSystemGmlNamespace()
             Assert.Equal(CoordinateSystem.Geography(1234), target.CoordinateSystem);
         }
 
+
```
Comment (gathogojr) [inline test/FunctionalTests/Microsoft.Spatial.Tests/GmlReaderTests.cs:29]: Two lines added inadvertently?

File: src/Microsoft.Spatial/Gml/GmlReader.cs [hunk 2]
```diff
@@ -237,6 +237,43 @@ private void ReadAttributes(bool expectSrsName)
...
+                    do
+                    {
+                        string attributeName = this.reader.LocalName;
+                        if (attributeName == GmlConstants.SrsDimension)
+                        {
+                            int value;
+                            string attributeValue = this.reader.Value;
+                            try
+                            {
+                                value = XmlConvert.ToInt32(attributeValue);
```
Comment (gathogojr) [inline src/Microsoft.Spatial/Gml/GmlReader.cs:259]:
```suggestion
                                return XmlConvert.ToInt32(attributeValue);

```

File: src/Microsoft.Spatial/Gml/GmlReader.cs [hunk 3]
```diff
@@ -708,6 +776,17 @@ private bool ReadStartOrEmptyElement(string element)
                 return !isEmptyElement;
             }
 
+            private int? ReadAttribute()
```
Comment (gathogojr) [inline src/Microsoft.Spatial/Gml/GmlReader.cs:779]: Curious where or when this is applied...

**Review by gathogojr — APPROVED** [review]
Left a few comments otherwise looks good to me

File: test/FunctionalTests/Microsoft.Spatial.Tests/GmlReaderTests.cs (hunk 1 above)
Comment (xuzhg) [reply test/FunctionalTests/Microsoft.Spatial.Tests/GmlReaderTests.cs:29]: Roll-backed.

File: src/Microsoft.Spatial/Gml/GmlReader.cs (hunk 2 above)
Comment (xuzhg) [reply src/Microsoft.Spatial/Gml/GmlReader.cs:259]: Roll-backed the unused codes

File: src/Microsoft.Spatial/Gml/GmlReader.cs (hunk 3 above)
Comment (xuzhg) [reply src/Microsoft.Spatial/Gml/GmlReader.cs:779]: Rollbacked

---
In reply to: [516195715](https://github.com/OData/odata.net/pull/1915#discussion_r516195715) [](ancestors = 516195715)

**Pull request merged**